kong/runloop/plugin_servers/process.lua

```diff
@@ -197,6 +202,38 @@ local function grab_logs(proc, name)
   end
 end
 
+-- if a plugin server is not ready after 20s of waiting we consider it failed
+local pluginserver_start_timeout = 20
+
+function proc_mgmt.connection_check_timer(premature, server_def)
+  if premature then
+    return
+  end
+
+  if ngx.config.subsystem ~= "http" then
```

Comment:
This should be a module level var, like `local is_http = ngx.config.subsystem ~= "http"`